lightning/src/ln/channel.rs

```diff
@@ -10299,11 +10459,101 @@ where
 
 		// Note on channel reserve requirement pre-check: as the splice acceptor does not contribute,
 		// it can't go below reserve, therefore no pre-check is done here.
-		// TODO(splicing): Once splice acceptor can contribute, add reserve pre-check, similar to the one in `splice_ack`.
 
-		// TODO(splicing): Store msg.funding_pubkey
-		// TODO(splicing): Apply start of splice (splice_start)
+		let pre_channel_value = self.funding.value_to_self_msat;
+		let _post_channel_value = PendingSplice::compute_post_value(
+			pre_channel_value,
+			their_funding_contribution_satoshis,
+			our_funding_contribution_satoshis,
+		);
+		let _post_balance = PendingSplice::add_checked(
+			self.funding.value_to_self_msat,
+			our_funding_contribution_satoshis,
+		);
+		// TODO: Early check for reserve requirement
+
+		Ok(())
+	}
+
+	/// See also [`validate_splice_init`]
+	#[cfg(splicing)]
+	pub(crate) fn splice_init<ES: Deref, L: Deref>(
+		&mut self, msg: &msgs::SpliceInit, our_funding_contribution_satoshis: i64,
+		signer_provider: &SP, entropy_source: &ES, holder_node_id: &PublicKey, logger: &L,
+	) -> Result<msgs::SpliceAck, ChannelError>
+	where
+		ES::Target: EntropySource,
+		L::Target: Logger,
+	{
+		let _res = self.validate_splice_init(msg)?;
+
+		let pre_channel_value = self.funding.get_value_satoshis();
+		let their_funding_contribution_satoshis = msg.funding_contribution_satoshis;
+
+		let post_channel_value = PendingSplice::compute_post_value(
+			pre_channel_value,
+			our_funding_contribution_satoshis,
+			their_funding_contribution_satoshis,
+		);
```

Comment:
We probably don't want this on `PendingSplice`. Feels like this should be computed inside `FundingScope::for_splice`, returning an error if any reserve checks fail. Maybe we move that call into `validate_splice_init` and have it return `FundingScope`?